cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,233 @@
+    Select your cloud provider:
+    > 1. Azure
+      2. AWS
+      3. [back]
+    ```
+
+1. Enter or use the Azure subscription and resource group name. Select  "Workload Identity" for the Azure cloud provider.
+    ```
+    Select the identity for the Azure cloud provider 
+    1. Service principal                               
+    > 2. Workload Identity    
+    ```
+
+1. Enter the Entra ID Application Client ID.
+    ```
+    Please follow the guidance at aka.ms/rad-workload-identity to set up workload identity for Radius.
+
+    Enter the Client ID of the Entra ID Application that you have created for Radius.
+
+    > Enter the Client ID:
+    ```
+
+Radius is now configured to use Workload Identity for authentication with Azure.
+
+#### User story 2: As an infrastructure operator, I can configure the Radius Azure Provider to use Workload Identity for authentication manually.
+
+1. Install the `rad` CLI.
+
+1. Install Radius using `rad install kubernetes`
+    ```
+    rad install kubernetes --set global.azureWorkloadIdentity.enabled=true
+    ```
+
+1. Create a Radius environment and add an Azure cloud provider.
+    ```
+    rad group create default
+    rad env create default
+    rad env update default --azure-subscription-id <subscription-id> --azure-resource-group <resource-group> 
+    ```
+
+1. Register the Entra ID Application using `rad credential register azure`.
+    ```
+    rad credential register azure --client-id <client-id>
+    ```
+
+Radius is now configured to use Workload Identity for authentication with Azure.
+
+## User Experience
+
+We will be adding a new option to the `rad init --full` command to allow the user to choose between service principal and Azure Workload Identity.
+
+We will be adding a new option to the Radius Helm chart to allow the user to enable Azure Workload Identity during installation via `rad install kubernetes --set global.azureWorkloadIdentity.enabled=true`.
+
+**Sample Input:**
+
+See above
+
+**Sample Output:**
+
+See above
+
+**Sample Recipe Contract:**
+
+n/a
+
+## Design
+
+### High Level Design
+
+During installation, if the user enables the Azure Workload Identity feature, the Radius control plane (bicep-de, ucp, applications-rp) pods will be annotated with the `azure.workload.identity/use: "true"` label. The AzWI mutating admission webhook will see this label on the pods and will project a signed service account token to each of the pods. The signed service account token and the credential registered by the user will be used by Radius to authenticate with Azure.
+
+Please see https://azure.github.io/azure-workload-identity/docs/concepts.html for more information on how the AzWI mutating admission webhook works.
+
+### Architecture Diagram
+
+![Architecture Diagram](./2024-04-azure-workload-identity/architecture.png)
+
+### Detailed Design
+
+#### Advantages (of each option considered)
+
+n/a
+
+#### Disadvantages (of each option considered)
+
+n/a
+
+#### Proposed Option
+
+n/a
+
```

Comment:
I might be a little confused about the purpose of this section. Let's talk about it if we have time.